### DISCUSSION TITLE
graph on Bill detail page

### DISCUSSION THREAD

**Discussion by JeroenED** [firefly-iii/firefly-iii#8852] [General]
Hi,

I use the bills a lot but something always bothered me. Why does my graph not look like it on the docs page?

In the docs the graph has a line on what it the current trend of the bill. If the line goes up, it it gets immediately clear the bill is getting more expensive.

This is not immediately clear to in the view I have which has bar for each entry of the bill. However it is quite clear to see which bills were over the maximum.

I did search a bit in the code and found that the line or bar is hardcoded and the view was changed in PR #3018 and #3016 for which I understand the arguments. The bar for minimum and maximum is quite some unnecessary clutter.

I want to propose these adjustments:
* All line graphs. So you can see the trend clearly while you can see which bill entries were over maximum. 
* Save the current maximum and minimum in the journal so the minimum and maximum can change overtime (eg. inflation can raise your internet bill)
* Add a line for the average so you can see the trend even more clearly.
* Make all lines (except the actual amount) hideable. So you can choose the clutter you want
* Additionally something I think that could be a improvement:  Add fixed rate bill. (eg. bills like office 365 is just a yearly cost of 89 euros no rate between a min and max depending on your usage)

Hope you like my suggestions.

https://github.com/firefly-iii/firefly-iii/blob/main/app/Http/Controllers/Chart/BillController.php#L136
https://docs.firefly-iii.org/images/how-to/firefly-iii/finances/bills-show.png

**Comment by JC5** [comment]
I do, and I will remember them for https://github.com/firefly-iii/firefly-iii/issues/6306
